### PR TITLE
fix(logging): drop log level for many log messages

### LINKED ISSUE #1002: (com.aws.greengrass.Nucleus): Reduce volume of INFO level logging
**Feature Description**
Thousands of log lines are written into the `greengrass.log` log file for even relatively simple operations such as deployments. The logging volume should be reduced to increase usability of the logs and reduce wear on embedded storage.

**Use Case**
Embedded solutions may often run on devices which not only are storage-constrained, but the storage may also have limited write cycles or poor wear leveling, such as is the case for inexpensive eMMC storage.

While logs are important for monitoring the status of the device and especially crucial for troubleshooting, and log rotation reduces the logs stored to a manageable amount, a large volume of logging will wear down embedded device storage faster.

**Proposed Solution**
Move all current INFO level logging to the DEBUG log level and only hand-pick the most useful log messages back to INFO.

**Other**

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by MikeDombo**
**Issue #, if available:**
Resolves #1002 

**Description of changes:**
Dropping log level from info to debug or trace when appropriate. This reduces the amount of logs that we and customer need to troll through when debugging. Logs are of course still available to debug or trace level if we need to dig deeper.

In our unit and integ tests run by GitHub this PR reduces the number of log lines by 25% from ~42,000 to 32,000.

**Why is this change necessary:**

**How was this change tested:**
Verified E2E tests locally

**Any additional information or context required to review the change:**

**Checklist:**
 - [ ] Updated the README if applicable
 - [ ] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] Updated or added new end-to-end tests
 - [ ] If your code makes a remote network call, it was tested with a proxy
 - [x] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/integrationtests/java/com/aws/greengrass/integrationtests/e2e/BaseE2ETestCase.java [hunk 1]
```diff
@@ -226,7 +228,11 @@ protected BaseE2ETestCase() throws Exception {
     }
 
     public static void setDefaultRunWithUser(Kernel kernel) {
-        new DeviceConfiguration(kernel).getRunWithDefaultPosixUser().dflt("nobody");
+        String user = "nobody";
+        if (Platform.getInstance() instanceof DarwinPlatform) {
+            user = System.getProperty("user.name");
```
Comment (tilo-chen) [inline src/integrationtests/java/com/aws/greengrass/integrationtests/e2e/BaseE2ETestCase.java:233]: Curious why Mac requires an actual valid user

**Review by tilo-chen — APPROVED** [review]

File: src/integrationtests/java/com/aws/greengrass/integrationtests/e2e/BaseE2ETestCase.java (hunk 1 above)
Comment (rbattle) [reply src/integrationtests/java/com/aws/greengrass/integrationtests/e2e/BaseE2ETestCase.java:233]: `nobody` is a valid user on the mac too - but the default mac sudoers file sets it up so you need to enter the current users password before switching

File: src/main/java/com/aws/greengrass/authorization/AuthorizationPolicyParser.java [hunk 2]
```diff
@@ -47,7 +47,7 @@
         Topics allServices = kernel.getConfig().findTopics(SERVICES_NAMESPACE_TOPIC);
 
         if (allServices == null) {
-            logger.atInfo("load-authorization-all-services-component-config-retrieval-error")
+            logger.atWarn("load-authorization-all-services-component-config-retrieval-error")
```
Comment (rbattle) [inline src/main/java/com/aws/greengrass/authorization/AuthorizationPolicyParser.java:50]: :+1:

File: src/main/java/com/aws/greengrass/builtin/services/lifecycle/LifecycleIPCEventStreamAgent.java [hunk 3]
```diff
@@ -236,9 +233,10 @@ public DeferComponentUpdateResponse handleRequest(DeferComponentUpdateRequest re
                 if (deferComponentUpdateRequestFuture == null) {
                     throw new ServiceError("Time limit to respond to PreComponentUpdateEvent exceeded");
                 } else {
+                    log.atDebug().log("Processing deployment deferral from {} for deployment {}", serviceName,
```
Comment (rbattle) [inline src/main/java/com/aws/greengrass/builtin/services/lifecycle/LifecycleIPCEventStreamAgent.java:236]: question: do we log somewhere that the deployment is deferred?

File: src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java [hunk 4]
```diff
@@ -361,7 +361,6 @@ private void setupCommWithIotJobs() {
                     logger.atWarn().log("Interrupted while subscribing to Iot Jobs topics");
                     return;
                 }
-                logger.atInfo().log("Connection established to IoT cloud");
```
Comment (rbattle) [inline src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java:364]: question: do we log somewhere when we are connected? In the MQTT client itself?

Comment (MikeDombo) [reply src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java:364]: Yeah, in the MQTT client.

File: src/main/java/com/aws/greengrass/builtin/services/lifecycle/LifecycleIPCEventStreamAgent.java (hunk 3 above)
Comment (MikeDombo) [reply src/main/java/com/aws/greengrass/builtin/services/lifecycle/LifecycleIPCEventStreamAgent.java:236]: Yes, https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/1171b2eb3e39917c1e9273d6d05d2f9eb344f8ba/src/main/java/com/aws/greengrass/lifecyclemanager/UpdateSystemPolicyService.java#L202

**Review by rbattle — APPROVED** [review]

**Pull request merged**